documentation/monitoring.md

```diff
@@ -0,0 +1,110 @@
+# Monitoring
+
+hwbench is using a specific monitoring engine to collect data from several sources :
+- BMC
+- PDU
+- Turbostat
+
+Regarding the source, power, frequencies, thermal or cooling metrics are collected.
+
+# Concept
+During each benchmark, and if the monitoring is enabled, metrics are collected every 2 seconds and aggregated every 10 seconds to get statistics over this period of time.
+
+At the end of the benchmark the monitoring metrics are added in the a result file. hwgraph will use them to plot how these components behave during the benchmark.
+
+# Usage
+To enable the monitoring feature, just set the `monitor` directive in the configuration file.
+
+As per this release, only `monitor=all` is supported. In future releases, it will be possible to list the source to be considered.
+
+# Configuration file
+When monitoring is enabled, the `-m <config_file>` option must be used to describe the server's configuration.
+This file is separated from the job file as it could be specific to each host.
+
+Each source is defined like the following :
+
+```
+  [section_name]
+  username=<username>
+  password=<password>
+  type=<type>
+ ```
+
+ ## BMC
+ When defining a BMC, the `type` must be set to `BMC`. The `section_name` could the hardware vendor name like `DELL`, `HPE`, or `default`.
+
+ A typical example looks like :
+ 
+ ```
+ #[HPE]
+username=Administrator
+password=YOURPASSWORD
+type=BMC
+ ```
+
+**Note**: there is no need to define the URL of the BMC, it will be automatically detected at runtime.
+
+A single BMC configuration will be used per server and vendor specific will be selected first if matching the running hardware.
+
+The BMC code is using `redfish` endpoints to monitor the server. Vendor specific endpoints can be used in addition of the generic ones to get all meaningful metrics.
+
+## PDU
+When defining a PDU,
+- a `section_name`, a user-defined value to represent this PDU
+- the ``type`` must be set to `PDU`.
+- a ``driver`` must be chosen
+- an ``URL`` is required
+- the ``outlet`` port must be selected
+
+ A typical example looks like :
+
+```
+[myPDU]
+username=admin
+password=admin
+type=PDU
+driver=raritan
+url=http://mypdu/
+outlet=21
+```
+
+**Note**: Several PDU configurations can be defined and used simultaneously.
+
+
+### Driver
+There exist many PDU providers and the software quality may vary a lot and so the protocols. To ensure a good compatibility with them, drivers can be added to hwbench.
+
+For this release, only **raritan** driver exists but as it uses some redfish endpoints, it might work on other products.
```

Comment:
I still wonder if it makes sense to call it raritan since it's a generic redfish driver (although it has only been tested with raritan PDUs). Maybe for later we can rename it, after we have tested more PDU brands;  but it might break existing configs then.

Reply:
My point was to name it Raritan until we get a 'proof' that this code is generic enough to be reused over multiple brands.
If so we reach this level of confidence, then we could move the code out of raritan and move it to generic.

Reply:
You are right, we don't know if it's generic. The main issue is how stable do we want the configs to be.